### DISCUSSION TITLE
Save connection in MQTTX Web

### DISCUSSION THREAD

**Discussion by brumela** [emqx/MQTTX#1566] [Q&A] [answered]
Is it possible to save connection in MQTTX Web?
My version is v1.9.8 running in docker.

**Comment by ysfscream** [comment]
Hi!

Since MQTTX Web is a browser-based application, it relies primarily on the storage capabilities provided by the browser to save this data. You can access your connection information in any browser that supports MQTTX Web if you use the same browser.

**Reply by brumela** [reply]
Ok, I see. 
But It's a little bit annoying to enter connection and subscription every time after browser is restarted.

**Reply by ysfscream** [reply, marked as answer]
Actually, MQTTX Web uses the browser's localStorage to save connection information, which should typically remain available even after restarting the browser. It's unusual that you need to re-enter the connection and subscription details each time. This issue might be due to your browser settings, particularly if it's set to clear cache or localStorage upon exiting, or due to certain browser extensions. Please check these settings or try using a different browser to see if the issue persists. Alternatively, you may want to consider using MQTTX Desktop, which offers more persistent storage for connection information. Let us know if this helps or if you continue to face the same issue.

**Reply by brumela** [reply]
Yes, you are right, I tested on different (non default) browser and after restart, connection and subscriptions are restored.
Thanks
